File: TestApp/app/src/main/java/com/romiope/testapp/network/RestClient.java
package com.romiope.testapp.network;

import com.romiope.testapp.network.v1.Api;

import retrofit2.Retrofit;
import retrofit2.adapter.rxjava.RxJavaCallAdapterFactory;
import retrofit2.converter.gson.GsonConverterFactory;

public class RestClient {

    static private RestClient instance = new RestClient();
    private final Api api;

    public Api getApi() {
        return api;
    }

    public static RestClient getInstance() {
        return instance;
    }

    public RestClient() {
        Retrofit.Builder builder = new Retrofit.Builder()
                .addCallAdapterFactory(RxJavaCallAdapterFactory.create())
                .addConverterFactory(GsonConverterFactory.create());

        api = new Api(builder);
    }
}

File: app/src/main/java/com/tesonet/testio/service/client/RestClient.kt
package com.tesonet.testio.service.client

import okhttp3.OkHttpClient
import okhttp3.OkHttpClient.Builder
import retrofit2.Retrofit
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory

class RestClient {

    fun createClient(): ApiClient {
        return getRetrofit(createRetrofitBuilder()).create(ApiClient::class.java)
    }

    private fun createRetrofitBuilder(): Retrofit.Builder {
        return Retrofit.Builder()
            .baseUrl(API_BASE_URL)
            .client(OkHttpClient().newBuilder().build())
            .addConverterFactory(GsonConverterFactory.create())
            .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
    }

    private fun getRetrofit(retrofitBuilder: Retrofit.Builder): Retrofit {
        return retrofitBuilder.client(Builder().build()).build()
    }

    companion object {
        const val API_BASE_URL: String = "https://playground.tesonet.lt"
    }
}